sunpy/map/mapbase.py

```diff
@@ -572,6 +572,27 @@ def max(self, *args, **kwargs):
         """
         return self.data.max(*args, **kwargs)
 
+    @property
+    def unit(self):
+        """
+        Unit of the map data.
+
+        This is taken from the 'BUNIT' FITS keyword. If no 'BUNIT' entry is
+        present in the metadata then this returns `None`. If the 'BUNIT' value
+        cannot be parsed into a unit a warning is raised, and `None` returned.
+        """
+        unit_str = self.meta.get('bunit', None)
+        if unit_str is None:
+            return
+        else:
+            unit = u.Unit(unit_str, parse_strict='warn')
```

Comment:
We should really be parsing these units with just the FITS unit convention?
```suggestion
            unit = u.Unit(unit_str, format="fits", parse_strict='warn')
```